app/src/main/java/io/appium/uiautomator2/handler/Swipe.java

```diff
@@ -1,51 +1,91 @@
 package io.appium.uiautomator2.handler;
 
+import android.support.test.uiautomator.UiDevice;
+import android.support.test.uiautomator.UiObjectNotFoundException;
+
 import org.json.JSONException;
 import org.json.JSONObject;
 
+import io.appium.uiautomator2.common.exceptions.InvalidCoordinatesException;
 import io.appium.uiautomator2.handler.request.SafeRequestHandler;
 import io.appium.uiautomator2.http.AppiumResponse;
 import io.appium.uiautomator2.http.IHttpRequest;
+import io.appium.uiautomator2.model.AndroidElement;
+import io.appium.uiautomator2.model.KnownElements;
 import io.appium.uiautomator2.server.WDStatus;
 import io.appium.uiautomator2.utils.Device;
 import io.appium.uiautomator2.utils.Logger;
+import io.appium.uiautomator2.utils.Point;
+import io.appium.uiautomator2.utils.PositionHelper;
 
 public class Swipe extends SafeRequestHandler {
+    UiDevice device;
 
     public Swipe(String mappedUri) {
         super(mappedUri);
     }
 
     @Override
     public AppiumResponse safeHandle(IHttpRequest request) {
+        Point absStartPos, absEndPos;
+        final boolean isSwipePerformed;
         try {
+            final SwipeArguments swipeArgs;
             JSONObject payload = getPayload(request);
-            boolean isActionPerformed;
-            String actionMsg;
-            Logger.info("Json Payload: ", payload.toString());
-
-            // TODO Swipe on Element
-
-            int startX = payload.getInt("startX");
-            int startY = payload.getInt("startY");
-            int endX = payload.getInt("endX");
-            int endY = payload.getInt("endY");
-            int steps = payload.getInt("steps");
+            Logger.info("JSON Payload : ", payload.toString());
+            swipeArgs = new SwipeArguments(request);
+            device = Device.getUiDevice();
```

Comment:
Any reason we kept **deivce** as instance variable? but looking at the code it seems we don't require it.